### PR TITLE
Copy-paste old arrow code into subdir to remove Arrow package dependency

### PR DESCRIPTION
**Pull request by quinnj**

### CODE REVIEW

**Comment by quinnj** [comment]
Hmmm, we seem to have bigger compat problems; the arrow code is doing `getlevels(x::CategoricalArray) = x.pool.index` which seems to not be valid in newer CategoricalArrays releases; but if we put upper bounds on CAs.jl, then that makes the DataFrames.jl bump not work. @nalimilan any advice?

**Comment by quinnj** [comment]
Ok, looks like we should just be calling `levels(c)`; so let's see how CI likes this

**Review by nalimilan — APPROVED** [review]
Thanks! Actually I hadn't seen your PR and I started applying the same fix here: https://github.com/JuliaData/Arrow.jl/pull/70. :-)

**Comment by nalimilan** [comment]
Failures on 32-bit Windows seem to be due to passing an `Int64` to `fieldtype`. That's weird since `Tables.columnindex` seems to always return an `Int`, but maybe just do `fieldtype(T, Int(col))`.

File: Manifest.toml [hunk 1]
```diff
@@ -0,0 +1,231 @@
+# This file is machine-generated - editing it directly is not advised
```
Comment (nalimilan) [inline Manifest.toml:1]: Did you intend to commit this file?

Comment (quinnj) [reply Manifest.toml:1]: nope!

**Comment by nalimilan** [comment]
32-bit Windows still fails, but looking at the AppVeyor log, the `fieldtype` failure has been there for at least one year so this PR doesn't make things worse.

**Comment by nalimilan** [comment]
Should we tag a release now?

**Comment by quinnj** [comment]
Tagged

**Pull request merged**